state/cloudcredentials.go

```diff
@@ -115,7 +129,21 @@ func (st *State) RemoveCloudCredential(tag names.CloudCredentialTag) error {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		return removeCloudCredentialOps(tag), nil
+		ops := removeCloudCredentialOps(tag)
+
+		cloudCredKey := cloudCredentialGlobalKey(tag)
+		if !force {
+			refcounts, closer := st.db().GetCollection(globalrefcountsC)
+			defer closer()
+			refOp, err := nsRefcounts.RemoveOp(refcounts, cloudCredKey, 0)
```

Comment:
This is going to return a pretty unhelpful error. Can you please check for `errRefcountChanged`, and return an error saying something like "cannot remove cloud credential %q, still in use by %d models"?